tests/unittests/Foundation/NSUndoManagerTests.mm

```diff
@@ -0,0 +1,239 @@
+//******************************************************************************
+//
+// Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <Foundation/Foundation.h>
+#import <TestFramework.h>
+
+@interface someObject : NSObject
+@property NSUndoManager* undoManager;
+@property NSNumber* value;
+@property NSMutableArray* arrayOfNumbers;
+- (id)initWithValue:(NSNumber*)num;
+- (id)initWithEmptyArray;
+- (void)updateNumber:(NSNumber*)value;
+- (void)updateNumberSingleUndoStep:(NSNumber*)value;
+- (void)addToMutableArray:(NSNumber*)value;
+- (void)removeFromMutableArray:(NSNumber*)value;
+@end
+
+@implementation someObject
+
+- (id)initWithValue:(NSNumber*)num {
+    self = [super init];
+    _undoManager = [[[NSUndoManager alloc] init] autorelease];
+    _value = num;
+    _arrayOfNumbers = nil;
+    return self;
+}
+
+- (id)initWithEmptyArray {
+    self = [super init];
+    _undoManager = [[[NSUndoManager alloc] init] autorelease];
+    _value = nil;
+    _arrayOfNumbers = [NSMutableArray array];
+    return self;
+}
+
+- (void)updateNumber:(NSNumber*)value {
+    [_undoManager registerUndoWithTarget:self selector:@selector(updateNumber:) object:_value];
+    _value = value;
```

Comment:
I would rather do something like, _value = newValue, to stay consistent with using the ivar directly instead of the getters/setters.

---
In reply to: [76114880](https://github.com/Microsoft/WinObjC/pull/900#discussion_r76114880) [](ancestors = 76114880)